pkg/resizer/csi_resizer.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resizer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/kubernetes-csi/external-resizer/pkg/csi"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+)
+
+// NewCSIResizer creates a new resizer responsible for resizing CSI volumes.
+func NewCSIResizer(address string, timeout time.Duration) (Resizer, error) {
```

Comment:
It is bit odd that all these functions take timeout parameter but convert this into a context anyways. Shouldn't we just be passing `Context` here in the first place?

Reply:
`WithTimeout` returns `WithDeadline(parent, time.Now().Add(timeout))`, so I think we need to create the context just before each CSI operation. If we passed a `Context` to `NewCSIResizer` method, it means the total time consumed by all CSI operations(`waitDriverReady`, `getDriverName`, etc.) should not exceed `timeout`. This is not what we want.